source/postprocess/heat_flux_map.cc

```diff
@@ -0,0 +1,221 @@
+      typename DoFHandler<dim>::active_cell_iterator
+      cell = this->get_dof_handler().begin_active(),
+      endc = this->get_dof_handler().end();
+
+      for (; cell!=endc; ++cell)
+        if (cell->is_locally_owned())
+          for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+            if (cell->at_boundary(f))
+              {
+                fe_face_values.reinit (cell, f);
+
+                //evaluate position of heatflow to write into output file
+                const Point<dim> midpoint_at_surface = cell->face(f)->center();
+
+                // get the various components of the solution, then
+                // evaluate the material properties there
+                fe_face_values[this->introspection().extractors.temperature].get_function_gradients (this->get_solution(),
+                    temperature_gradients);
+
+                for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                  fe_face_values[this->introspection().extractors.compositional_fields[c]].get_function_values(this->get_solution(), composition_values[c]);
+
+
+                for (unsigned int i=0; i<fe_face_values.n_quadrature_points; ++i)
+                  {
+                    for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                      in.composition[i][c] = composition_values[c][i];
+                  }
+
+                this->get_material_model().evaluate(in, out);
+
+
+                // Calculate the normal conductive heat flux given by the formula
+                //   j = - k * n . grad T
+
+                double normal_flux = 0;
+                for (unsigned int q=0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    const double thermal_conductivity
+                      = out.thermal_conductivities[q];
+
+                    normal_flux += -thermal_conductivity *
+                                   (temperature_gradients[q] * fe_face_values.normal_vector(q)) *
+                                   fe_face_values.JxW(q);
+
+                  }
+
+                //store final position and heatflow
+                stored_values.push_back (std::make_pair(midpoint_at_surface, normal_flux));
+
+              }
+
+
+      // Write the solution to an output file
+      for (unsigned int i=0; i<stored_values.size(); ++i)
+        {
+          output << stored_values[i].first
+                 << ' '
+                 << stored_values[i].second
+                 << std::endl;
+        }
+
+
+
+      const std::string filename = this->get_output_directory() +
+                                   "heat_flux." +
+                                   Utilities::int_to_string(this->get_timestep_number(), 5);
+
+      const unsigned int max_data_length = Utilities::MPI::max (output.str().size()+1,
+                                                                this->get_mpi_communicator());
+      const unsigned int mpi_tag = 123;
```

Comment:
I suspect that that's the same tag as for the other plugin. choose a different number. any number will do.